sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy

```diff
@@ -144,7 +145,12 @@ class SentryPlugin implements Plugin<Project> {
      * @return
      */
     static String getDebugMetaPropPath(Project project, ApplicationVariant variant) {
-        return "${variant.mergeAssets.outputDir}/sentry-debug-meta.properties"
+        try {
+            return "${variant.mergeAssetsProvider.outputDir}/sentry-debug-meta.properties"
+        } catch (Throwable ignored) {
+            return "${variant.mergeAssets.outputDir}/sentry-debug-meta.properties"
```

Comment:
`outputDir` has type `Provider<Directory>` at least at agp 3.4. 
Therefore this function returns somethinf like this `property(interface org.gradle.api.file.Directory, map(property(interface org.gradle.api.file.Directory, fixed(class org.gradle.api.internal.file.DefaultFilePropertyFactory$FixedDirectory, /Users/user/projects/myAwesomeApp/app/build))))/sentry-debug-meta.properties`

I believe it should be `return variant.mergeAssets.outputDir.get().file("sentry-debug-meta.properties").getAsFile().path

Reply:
What @nesterov-n said. In general I'd favor an implementation where instead of try-catching possible exceptions you rather check for the expected type and then call the right method. This - together with a couple of invariants / assertions in the right places - will result in a much more consistent and stable implementation that even gives you proper diagnostics in case something will change (again). And if we learned one thing with the AGP then that change is a constant thing.